heka/sandbox/decoders/extract_telemetry_dimensions.lua

```diff
@@ -12,13 +12,29 @@ local msg = {
 Timestamp   = nil,
 Type        = "telemetry",
 Payload     = nil,
-Fields      = { sourceName = "telemetry" }
+EnvVersion  = 1
 }
 
 local UNK_DIM = "UNKNOWN"
 local UNK_GEO = "??"
+local environment_objects = {"build", "settings", "profile", "partner", "system", "addons"}
+local main_ping_objects = {"info", "simpleMeasurements", "histograms", "keyedHistograms", "chromeHangs", "threadHangStats", "log", "fileIOReports", "lateWrites", "addonDetails", "addonHistograms", "UIMeasurements", "slowSQL", "slowSQLstartup", "childPayloads"}
+
+local function split_objects(root, section, objects)
+    if type(root) ~= "table" then return end
+
+    for i, name in ipairs(objects) do
+        if type(root[name]) == "table" then
+            local ok, json = pcall(cjson.encode, root[name])
+            if ok then
+                msg.Fields[string.format("%s.%s", section, name)] = json
```

Comment:
Should we check to make sure we're not overwriting an existing field name? Given the `section.name` scheme, that is unlikely, but it may be good to detect that case.